### PR TITLE
client: moved from zigzag

### PR DESCRIPTION
**Pull request by dimuls**

### CODE REVIEW

File: bitlum.go [hunk 1]
```diff
@@ -0,0 +1,763 @@
...
+)
+
+const exchangeID = "bitlum"
+
+// Client is the http://exchange.bitlum.io exchange
+type Client struct {
+	backend
+}
+
+// NewExchangeClient create new bitlum exchange with specified URL and
```
Comment (avemeva) [inline bitlum.go]: NewExchangeClient => NewClient

File: bitlum.go [hunk 2]
```diff
@@ -0,0 +1,763 @@
...
+func NewClient(url string, authToken string) *Client {
+	return &Client{
+		backend: &graphqlBackend{
+			url:       url,
+			authToken: authToken,
+		},
+	}
+}
+
+// ID returns exchange ID
```
Comment (avemeva) [inline bitlum.go]: What is exchange id from external user standpoint?

File: bitlum.go [hunk 3]
```diff
@@ -0,0 +1,763 @@
...
+
+	if err := resp.Error(); err != nil {
+		return "", errors.New("bitlum server-side error: " +
+			err.Error())
+	}
+
+	return resp.Data.User.ID, nil
+}
+
+type Ticker struct {
```
Comment (avemeva) [inline bitlum.go]: What is ticker?

File: bitlum.go [hunk 4]
```diff
@@ -0,0 +1,763 @@
...
+	Market     string
+	Last       decimal.Decimal
+	ChangeLast decimal.Decimal
```
Comment (avemeva) [inline bitlum.go]: Add comments to fields

File: bitlum.go [hunk 5]
```diff
@@ -0,0 +1,763 @@
...
+
+	if err := resp.Error(); err != nil {
+		return nil, errors.New("bitlum server-side error: " +
+			err.Error())
+	}
+
+	return resp.Data.Markets, nil
+}
+
+type Ask struct {
```
Comment (avemeva) [inline bitlum.go]: Add comment

File: bitlum.go [hunk 6]
```diff
@@ -0,0 +1,763 @@
...
+	Price  decimal.Decimal
+	Volume decimal.Decimal
+}
+
+type Bid struct {
```
Comment (avemeva) [inline bitlum.go]: Add comment

File: bitlum.go [hunk 7]
```diff
@@ -0,0 +1,763 @@
...
+
+// depositRequestVariables is a query variables used in request
+// in bitlum Deposits method.
+type depositRequestVariables struct {
+	Assets []string `json:"assets"`
+	Offset int64    `json:"offset"`
+	Limit  int64    `json:"limit"`
+}
+
+type Deposit struct {
```
Comment (avemeva) [inline bitlum.go]: Add comment

File: bitlum.go [hunk 8]
```diff
@@ -0,0 +1,763 @@
...
+// createOrderRequestVariables is a query variables used in request
+// in bitlum CreateOrder method.
+type createOrderRequestVariables struct {
+	Market string          `json:"market"`
+	Amount decimal.Decimal `json:"amount"`
+}
+
+// CreateBidOrder creates bid order on market. Bid order means that
+// market[0] asset is used to buy market[1] asset
+func (c *Client) CreateOrder(market string,
```
Comment (avemeva) [inline bitlum.go]: Please rephrase the comment, would be better to get explanation from investopedia.

File: bitlum.go [hunk 9]
```diff
@@ -0,0 +1,763 @@
...
+	BlockHeight uint32
+
+	// The node's current view of the hash of the best block
+	BlockHash string
+
+	// Whether the wallet's view is synced to the main chain
+	SyncedToChain bool
+
+	// Whether the current node is connected to testnet
+	Testnet bool
```
Comment (avemeva) [inline bitlum.go]: We have to use golang version of comments "[Name] [explanation]"

File: backend.go [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// backend is client core for bitlum exchange service.
+type backend interface {
```
Comment (avemeva) [inline backend.go]: Explain why we need this interface, that is hard to understand for external viewer

**Review by avemeva — CHANGES_REQUESTED** [review]

File: bitlum.go (hunk 1 above)
Comment (dimuls) [reply bitlum.go]: -maybe just `New`?- ok got it

File: bitlum.go (hunk 2 above)
Comment (dimuls) [reply bitlum.go]: This ID can be used to distinguish exchanges in zigzag. But in this client it doesn't need. Will remove.

File: bitlum.go (hunk 3 above)
Comment (dimuls) [reply bitlum.go]: Its from https://en.wikipedia.org/wiki/Ticker_tape canonical name of stock exchange prices. Other exchanges like https://poloniex.com/support/api/ use the same name.

Comment (avemeva) [reply bitlum.go]: yeah, I meat lets add comment

File: bitlum.go (hunk 1 above)
Comment (avemeva) [reply bitlum.go]: yes, could choise

**Comment by avemeva** [comment]
Good, thanks for the PR, I think that is okay for first version of this client, but in future we would need make it even better : )

**Pull request merged**